### PR TITLE
add CyclicalCosineDecay

### PR DESCRIPTION
**Pull request by littletomatodonkey**
1. Add CyclicalCosineDecay. The strategy can be seen in : https://arxiv.org/pdf/2012.12645.pdf. In ocr, without any fine tune on hyper-parameters, it can directly takes a 0.3% accuracy gain for CRNN(with 10% more training epochs).
2. modify the epoch num, because if we want to save models per 5epochs and total epoch is set as 50, then the 50th epoch  model will not be saved for the code before(the last epoch is actually 49).

### CODE REVIEW

File: ppocr/optimizer/lr_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# copyright (c) 2020 PaddlePaddle Authors. All Rights Reserve.
```
Comment (WenmuZhou) [inline ppocr/optimizer/lr_scheduler.py:1]: 这个文件不需要吧

Comment (littletomatodonkey) [reply ppocr/optimizer/lr_scheduler.py:1]: 类似于框架中的lr_scheduler.py，和ocr中封装的learning_rate分开，感觉会更清楚一点

**Review by LDOUBLEV — APPROVED** [review]

**Review by WenmuZhou — APPROVED** [review]
